### PR TITLE
test: Move My Create To Platform: Fixing Tests

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 3396dd3</samp>

This pull request improves the functionality and testing of the `MyCreateReportPage` component, which allows users to create reports from selected expenses. It uses the new platform model and service methods for expenses, handles edge cases, and avoids array mutation. It also adds mock data for different types of expenses.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 3396dd3</samp>

> _To test `MyCreateReportPage`_
> _We added some mock data on stage_
> _We refactored the code_
> _To handle each mode_
> _And used new methods to gauge_

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 3396dd3</samp>

*  Add mock data for ready to report and non-reimbursable expenses ([link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-24d88c263ca5d224aa35b783df4640316640e6db32915015b6cce6ff83b7a05bR410-R427))
  - Import `ExpensesService` from the platform module and declare a spy object for it ([link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L12-R17), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L30-R31), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03R43), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03R60), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03R110-R113), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03R129))
  - Rename `selectedTxnIds` to `selectedExpenseIDs` and use optional chaining for `expenseIDs` ([link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L154-R161), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L162-R169), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L370-R373), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L379-R382), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L117-R118), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L126-R132))
  - Update the methods for sending the first report created event, getting the report title, and creating a report or draft to use the new method for getting the total amount of selected expenses ([link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L173-R193), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L192-R207), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L203-R222), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L318-R335), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L349-R354), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L89-R91), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L188-R189))
  - Remove the unused method and import of `getVendorDetails` and the related test cases and spies ([link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L12-R17), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L275-R303), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L298-R313), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L387-R393))
  - Add a new method for getting the total amount of selected expenses that are reimbursable and the related test cases ([link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L275-R303), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794R268-R271))
  - Update the method for getting all expenses to use the new service method, assign the expenses to `selectedElements`, and filter out the expenses that are not in `selectedExpenseIDs` ([link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L226-R234), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L243))
  - Reset the `selectedElements` array to an empty array when the report title input is changed ([link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794R210))
  - Add optional chaining to the previous expense spent at date to handle the case when it is undefined ([link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L252-R251))
  - Update the test cases for `selectExpense`, `toggleSelectAll`, and `ngOnInit` to use the new mock data and service method ([link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L306-R321), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L318-R335), [link](https://github.com/fylein/fyle-mobile-app/pull/2577/files?diff=unified&w=0#diff-e99cd7c1b24fdd0e0242661ffed035227ad84a86d656e6c34dd5e3429a700a03L387-R393))

## Clickup
https://app.clickup.com/t/86ctv3kkf

## Code Coverage
`100% Statements 114/114`
`92.72% Branches 51/55`
`100% Functions 37/37`
`100% Lines 103/103`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/mock-data/platform/v1/expense.data.ts [hunk 1]
```diff
@@ -407,3 +407,21 @@ export const perDiemExpenseWithMultipleNumDays: Expense = {
...
+    id: 'tx3rb9ZcrTRn',
+  },
+  {
+    ...expenseData,
+    amount: 100,
+    id: 'txWjW9qT2Vf1',
+  },
+];
+
+export const nonReImbursableExpense: Expense = {
```
Comment (Dimple16) [inline src/app/core/mock-data/platform/v1/expense.data.ts:424]: `nonReimbursableExpense`

File: src/app/fyle/my-create-report/my-create-report.page.ts [hunk 2]
```diff
@@ -123,12 +124,12 @@ export class MyCreateReportPage implements OnInit {
...
             tap(() =>
               this.trackingService.createReport({
-                Expense_Count: expenseIDs.length,
+                Expense_Count: expenseIDs?.length,
                 Report_Value: this.selectedTotalAmount,
               })
             ),
             switchMap((report: ReportV1) => {
-              if (expenseIDs.length > 0) {
+              if (expenseIDs?.length > 0) {
```
Comment (Dimple16) [inline src/app/fyle/my-create-report/my-create-report.page.ts:132]: `if (expenseIDs?.length)` should do

**Review by Dimple16 — APPROVED** [review]
minor comments

File: src/app/core/mock-data/platform/v1/expense.data.ts [hunk 3]
```diff
@@ -407,3 +407,21 @@ export const perDiemExpenseWithMultipleNumDays: Expense = {
   category: perDiemCategory,
   per_diem_num_days: 3,
 };
+
+export const readyToReportExpensesData: Expense[] = [
+  {
+    ...expenseData,
```
Comment (OmkarJ13) [inline src/app/core/mock-data/platform/v1/expense.data.ts:413]: I am also making few changes in expenseData, this [PR](https://github.com/fylein/fyle-mobile-app/pull/2561/files) will be merged soon.
Can you check if this has any conflicts

File: src/app/fyle/my-create-report/my-create-report.page.spec.ts [hunk 4]
```diff
@@ -170,13 +177,20 @@ describe('MyCreateReportPage', () => {
...
     reportService.getMyReportsCount.and.returnValue(of(0));
-    component.readyToReportEtxns = cloneDeep(selectedExpenses);
+    spyOn(component, 'getTotalSelectedExpensesAmount').and.returnValue(150);
+    component.readyToReportExpenses = [...readyToReportExpensesData, expenseData];
+    component.selectedElements = readyToReportExpensesData;
     fixture.detectChanges();
 
     component.sendFirstReportCreated();
-    tick(500);
+    tick(1000);
```
Comment (OmkarJ13) [inline src/app/fyle/my-create-report/my-create-report.page.spec.ts:186]: Didn't follow this change, do we need to increase the tick timer?

**Review by OmkarJ13 — APPROVED** [review]

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**